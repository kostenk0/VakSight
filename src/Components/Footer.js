import React from 'react'

const footerStyle = {
    backgroundColor: "#31373e",
    fontSize: "20px",
    color: "white",
    borderTop: "1px solid #E7E7E7",
    textAlign: "center",
    padding: "20px",
    position: "fixed",
    left: "0",
    bottom: "0",
    height: "60px",
    width: "100%"
  };
  const phantomStyle = {
    display: "block",
    padding: "20px",
    height: "60px",
    width: "100%"
  };
  function Footer1({ children }) {
    return (
      <div>
        <div style={phantomStyle} />
        <div style={footerStyle}>{children}</div>
      </div>
    );
  }

const Footer = () => {
    return (
        <div>
        <Footer1>
          <span>Допоможіть: 5168 7573 3228 5974</span>
        </Footer1>
      </div>
    );
};
export default Footer;