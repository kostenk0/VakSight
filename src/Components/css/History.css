.history-main{
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.history-div{
    padding-top: 0.5%;
    padding-right: 3%;
}