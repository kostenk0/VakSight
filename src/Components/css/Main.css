.main-div{
    padding: 10px;
}
.noMargin{
    margin-left: 0;
    margin-right: 0;
}
.card-div{
    background: yellow;
    margin-right: 5%;
    margin-left: 5%;
    margin-top:3%;
    margin-bottom: 3%;
}
.author-tabs-div{
    padding-top: 7px;
}
.select-div{
    margin-bottom: 9px;
}
.load-div{
    text-align: center;
}
.info-div{
    background: gainsboro;
    text-align: center;
    margin:3%;
}