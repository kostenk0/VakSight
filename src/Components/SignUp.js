import React, { Component } from 'react'
import { Form, Button } from 'react-bootstrap'
import './css/SignIn.css'
import Server from '../Services/Server'
import Header from './Header'

export default class SignUp extends Component {
    constructor(props) {
        super(props);
        this.sendRequest = this.sendRequest.bind(this);
        this.getEmail = this.getEmail.bind(this);
        this.getPassword = this.getPassword.bind(this);

        this.state = {
            email: '',
            password: '',
            loading: false
        };
    }
    getEmail(e) {
        this.setState({
            email: e.target.value
        });
    }
    getPassword(e) {
        this.setState({
            password: e.target.value
        });
    }
    sendRequest(e) {
        e.preventDefault();
        if (this.state.email !== '' && this.state.password.length >= 6) {
            this.setState({
                loading: true
            });
            const userData = {
                "email": this.state.email,
                "password": this.state.password
            };
            Server.signUp(userData).then(res => {
                this.setState({ loading: false });
                this.props.history.push('/signIn');
            });
        } else {
            alert('Некоректні дані. Спробуйте ще раз');
            this.setState({
                email: '',
                password: ''
            });
        };
    }

    render() {
        return (
            <div >
                <Header />
                <div class='signIn-main-div'>
                    <Form>
                        <h1>Реєстрація</h1>
                        <Form.Group controlId="formBasicEmail">
                            <Form.Label>Електронна пошта</Form.Label>
                            <Form.Control type="email" placeholder="Введіть email" value={this.state.email} onChange={this.getEmail} />
                            <Form.Text className="text-muted">
                                Ми нікому не передаємо Ваші дані.
                        </Form.Text>
                        </Form.Group>

                        <Form.Group controlId="formBasicPassword">
                            <Form.Label>Пароль</Form.Label>
                            <Form.Control type="password" placeholder="Довжина від 6 символів" value={this.state.password} onChange={this.getPassword} />
                            <Form.Text className="text-muted">
                                Якщо Ви вже зареєстровані натисність <a href='/signIn' style={{ color: 'blue' }}>ТУТ</a>
                            </Form.Text>
                        </Form.Group>
                        <Button variant="primary" type="submit" disabled={this.state.loading} onClick={!this.state.loading ? this.sendRequest : null}>
                            {this.state.loading ? 'Реєструємо...' : 'Підтвердити'}
                        </Button>
                    </Form>
                </div>
            </div>
        );
    }

}