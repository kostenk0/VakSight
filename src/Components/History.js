import React, { Component } from 'react'
import './css/History.css'
import { Card} from 'react-bootstrap'
import HeaderForAuth from './HeaderForAuth'
import Server from '../Services/Server'
import './css/Main.css'

export default class History extends Component {
    constructor(props) {
        super(props);
        this.state = {
            items: ""
        };
    }
    componentDidMount() {
        Server.getHistory(localStorage.getItem('user'), localStorage.getItem('email')).then(res => {
            this.setState({
                items: res
            });
        })
    }

    render() {
        return (<div>
            <HeaderForAuth />
            <div>{
                
                this.state.items
                    ? this.state.items.map((item) =>
                        <Card className="card-div">
                            <Card.Body>{item.content}</Card.Body>
                        </Card>
                    )
                    : <div className="load-div"><h1>Відправляємо голуба. Зачекайте...</h1></div>
            }</div>
        </div>);
    }
}