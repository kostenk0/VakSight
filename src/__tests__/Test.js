import React from 'react'
import SignIn from '../Components/SignIn'
import SignUp from '../Components/SignUp'
import Main from '../Components/Main'
import History from '../Components/History'
import Footer from '../Components/Footer'
import AuthMain from '../Components/AuthMain'

import renderer from 'react-test-renderer'

test('SignIn snapshot test', () => {
    const tree = renderer.create(
        <SignIn />
    );
    expect(tree).toMatchSnapshot();
});
test('SignUp snapshot test', () => {
    const tree = renderer.create(
        <SignUp />
    );
    expect(tree).toMatchSnapshot();
});
test('Main snapshot test', () => {
    const tree = renderer.create(
        <Main />
    );
    expect(tree).toMatchSnapshot();
});
test('History snapshot test', () => {
    const tree = renderer.create(
        <History />
    );
    expect(tree).toMatchSnapshot();
});
test('AuthMain snapshot test', () => {
    const tree = renderer.create(
        <AuthMain />
    );
    expect(tree).toMatchSnapshot();
});
test('Footer snapshot test', () => {
    const tree = renderer.create(
        <Footer />
    );
    expect(tree).toMatchSnapshot();
});