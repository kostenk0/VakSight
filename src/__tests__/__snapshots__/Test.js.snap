// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthMain snapshot test 1`] = `
<div>
  <div
    className="main-div"
  >
    <div
      className="flex-column nav nav-pills nav-fill"
      onKeyDown={[Function]}
    >
      <div
        className="nav-item"
      >
        <a
          className="nav-link active"
          data-rb-event-key="link-1"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Книга
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="nav-link"
          data-rb-event-key="link-2"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Електронне джерело
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="nav-link"
          data-rb-event-key="link-3"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Частина книги
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="nav-link"
          data-rb-event-key="link-4"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Дисертація
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="nav-link"
          data-rb-event-key="link-5"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Автореферат дисертації
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="nav-link"
          data-rb-event-key="link-6"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Періодичне видання
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Footer snapshot test 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark fixed-bottom"
>
  <span
    className="navbar-brand"
  >
    Розроблено студентами
  </span>
  <button
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="justify-content-end navbar-collapse collapse"
  >
    <span
      className="navbar-text"
    >
      Допоможіть: 5168 7573 3228 5974
    </span>
  </div>
</nav>
`;

exports[`History snapshot test 1`] = `
<div>
  <div
    className="history-main"
  >
    <ul>
      <div
        className="history-div"
      >
        <div
          className="card bg-light"
        >
          <div
            className="card-body"
          >
            Матяш І. Б. Діяльність Надзвичайної дипломатичної місії УНР в Угорщині : історія, спогади, арх. док. / І. Матяш, Ю. Мушка. – К. : Києво-Могилян. акад., 2005. – 397, [1] с. – (Бібліотека наукового щорічника "Україна дипломатична" ; вип. 1).
          </div>
        </div>
      </div>
      <div
        className="history-div"
      >
        <div
          className="card bg-light"
        >
          <div
            className="card-body"
          >
            Ромовська З. В. Сімейне законодавство України / З. В. Ромовська, Ю. В. Черняк. – К. : Прецедент, 2006. – 93 с. – (Юридична бібліотека. Бібліотека адвоката) (Матеріали до складання кваліфікаційних іспитів для отримання Свідоцтва про право на заняття адвокатською діяльністю ; вип. 11).
          </div>
        </div>
      </div>
      <div
        className="history-div"
      >
        <div
          className="card bg-light"
        >
          <div
            className="card-body"
          >
            Історія Свято-Михайлівського Золотоверхого монастиря / [авт. тексту В. Клос]. – К. : Грані-Т, 2007. – 119 с. – (Грані світу).
          </div>
        </div>
      </div>
      <div
        className="history-div"
      >
        <div
          className="card bg-light"
        >
          <div
            className="card-body"
          >
            Воскресіння мертвих : українська барокова драма : антологія / [упорядкув., ст., пер. і прим. В. О. Шевчук]. – К. : Грамота, 2007. – 638, [1] с.
          </div>
        </div>
      </div>
      <div
        className="history-div"
      >
        <div
          className="card bg-light"
        >
          <div
            className="card-body"
          >
            Розподіл населення найбільш численних національностей за статтю та віком, шлюбним станом, мовними ознаками та рівнем освіти [Електронний ресурс] : за даними Всеукр. перепису населення 2001 р. / Держ. ком. статистики України ; ред. О. Г. Осауленко. – К. : CD-вид-во "Інфодиск", 2004. – 1 електрон. опт. диск (CD-ROM) : кольор. ; 12 см. – (Всеукр. перепис населення, 2001). – Систем. вимоги: Pentium-266 ; 32 Mb RAM ; CD-ROM Windows 98/2000/NT/XP. – Назва з титул. екрану.
          </div>
        </div>
      </div>
      <div
        className="history-div"
      >
        <div
          className="card bg-light"
        >
          <div
            className="card-body"
          >
            Матяш І. Б. Діяльність Надзвичайної дипломатичної місії УНР в Угорщині : історія, спогади, арх. док. / І. Матяш, Ю. Мушка. – К. : Києво-Могилян. акад., 2005. – 397, [1] с. – (Бібліотека наукового щорічника "Україна дипломатична" ; вип. 1).
          </div>
        </div>
      </div>
      <div
        className="history-div"
      >
        <div
          className="card bg-light"
        >
          <div
            className="card-body"
          >
            Ромовська З. В. Сімейне законодавство України / З. В. Ромовська, Ю. В. Черняк. – К. : Прецедент, 2006. – 93 с. – (Юридична бібліотека. Бібліотека адвоката) (Матеріали до складання кваліфікаційних іспитів для отримання Свідоцтва про право на заняття адвокатською діяльністю ; вип. 11).
          </div>
        </div>
      </div>
      <div
        className="history-div"
      >
        <div
          className="card bg-light"
        >
          <div
            className="card-body"
          >
            Історія Свято-Михайлівського Золотоверхого монастиря / [авт. тексту В. Клос]. – К. : Грані-Т, 2007. – 119 с. – (Грані світу).
          </div>
        </div>
      </div>
      <div
        className="history-div"
      >
        <div
          className="card bg-light"
        >
          <div
            className="card-body"
          >
            Воскресіння мертвих : українська барокова драма : антологія / [упорядкув., ст., пер. і прим. В. О. Шевчук]. – К. : Грамота, 2007. – 638, [1] с.
          </div>
        </div>
      </div>
      <div
        className="history-div"
      >
        <div
          className="card bg-light"
        >
          <div
            className="card-body"
          >
            Розподіл населення найбільш численних національностей за статтю та віком, шлюбним станом, мовними ознаками та рівнем освіти [Електронний ресурс] : за даними Всеукр. перепису населення 2001 р. / Держ. ком. статистики України ; ред. О. Г. Осауленко. – К. : CD-вид-во "Інфодиск", 2004. – 1 електрон. опт. диск (CD-ROM) : кольор. ; 12 см. – (Всеукр. перепис населення, 2001). – Систем. вимоги: Pentium-266 ; 32 Mb RAM ; CD-ROM Windows 98/2000/NT/XP. – Назва з титул. екрану.
          </div>
        </div>
      </div>
    </ul>
  </div>
</div>
`;

exports[`Main snapshot test 1`] = `
<div>
  <div
    className="main-div"
  >
    <div
      className="flex-column nav nav-pills nav-fill"
      onKeyDown={[Function]}
    >
      <div
        className="nav-item"
      >
        <a
          className="nav-link active"
          data-rb-event-key="link-1"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Книга
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="nav-link"
          data-rb-event-key="link-2"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Електронне джерело
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="nav-link"
          data-rb-event-key="link-3"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Частина книги
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="nav-link"
          data-rb-event-key="link-4"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Дисертація
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="nav-link"
          data-rb-event-key="link-5"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Автореферат дисертації
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="nav-link"
          data-rb-event-key="link-6"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Періодичне видання
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`SignIn snapshot test 1`] = `
<div>
  <div
    class="signIn-main-div"
  >
    <form
      className=""
    >
      <h1>
        Вхід
      </h1>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formBasicEmail"
        >
          Електронна пошта
        </label>
        <input
          className="form-control"
          id="formBasicEmail"
          onChange={[Function]}
          placeholder="Введіть email"
          type="email"
          value=""
        />
        <small
          className="text-muted form-text"
        >
          Ми нікому не передаємо Ваші дані.
        </small>
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formBasicPassword"
        >
          Пароль
        </label>
        <input
          className="form-control"
          id="formBasicPassword"
          onChange={[Function]}
          placeholder="Введіть пароль"
          type="password"
          value=""
        />
        <small
          className="text-muted form-text"
        >
          Якщо Ви не зареєстровані натисність 
          <a
            href="/signUp"
            style={
              Object {
                "color": "blue",
              }
            }
          >
            ТУТ
          </a>
        </small>
      </div>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Підтвердити
      </button>
    </form>
  </div>
</div>
`;

exports[`SignUp snapshot test 1`] = `
<div>
  <div
    class="signIn-main-div"
  >
    <form
      className=""
    >
      <h1>
        Реєстрація
      </h1>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formBasicEmail"
        >
          Електронна пошта
        </label>
        <input
          className="form-control"
          id="formBasicEmail"
          onChange={[Function]}
          placeholder="Введіть email"
          type="email"
          value=""
        />
        <small
          className="text-muted form-text"
        >
          Ми нікому не передаємо Ваші дані.
        </small>
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formBasicPassword"
        >
          Пароль
        </label>
        <input
          className="form-control"
          id="formBasicPassword"
          onChange={[Function]}
          placeholder="Введіть пароль"
          type="password"
          value=""
        />
        <small
          className="text-muted form-text"
        >
          Якщо Ви вже зареєстровані натисність 
          <a
            href="/signIn"
            style={
              Object {
                "color": "blue",
              }
            }
          >
            ТУТ
          </a>
        </small>
      </div>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Підтвердити
      </button>
    </form>
  </div>
</div>
`;
